server/executor/poller_executor.go

```diff
@@ -159,7 +159,12 @@ func (pe DefaultPollerExecutor) donePollingTraces(job *PollingRequest, traceDB t
 		return false
 	}
 
-	if len(trace.Flat) > traceDB.MinSpanCount() && len(trace.Flat) == len(job.run.Trace.Flat) {
+	haveNotCollectedSpansSinceLastPoll := len(trace.Flat) == len(job.run.Trace.Flat)
+
+	// Today we consider that we finished collecting traces
+	// if we haven't collected any new spans since our last poll
+
+	if haveNotCollectedSpansSinceLastPoll {
```

Comment:
It's important to have the `len(trace.Flat) > 0` because we might not get any trace data for a long time. Usually, it takes more to receive the initial spans than to get the subsequent spans. That condition was important so we didn't stop polling before getting any span.